### PR TITLE
[MM-22876] - tooltip fix rhs sidebar

### PR DESCRIPTION
**Pull request by adityayavagal**
#### Summary
This pull request shows tooltips when right sidebar buttons

#### Ticket Link
Github: Fixes https://github.com/mattermost/mattermost-server/issues/14056
JIRA: https://mattermost.atlassian.net/browse/MM-22876

#### Related Pull Requests

- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

### CODE REVIEW

**Comment by adityayavagal** [comment]
Hi, I have updated the PR can you check once.

**Comment by sudheerDev** [comment]
Would you be interested in adding test cases.

**Comment by adityayavagal** [comment]
Actually I'm new to this so I could with little guidance from someone

**Comment by adityayavagal** [comment]
> Would you be interested in adding test cases.

Actually I'm new to this so I could with little guidance from someone

**Comment by adityayavagal** [comment]
Hello, should change or add anything to this PR because I see it's not yet merged.

**Comment by sudheerDev** [comment]
> Would you be interested in adding test cases.

Hey, I can help you with that. Would you mind pinging me on https://community.mattermost.com/? You can find me by the username @sudheerDev

**Comment by adityayavagal** [comment]
> > Actually I'm new to this so I could with little guidance from someone
> 
> Hey, I can help you with that. Would you mind pinging me on https://community.mattermost.com/? You can find me by the username @sudheerDev

Alright sure thank you

**Comment by sudheerDev** [comment]
> Hello, should change or add anything to this PR because I see it's not yet merged.

Sorry, we need to wait for @hmhealey review and then another QA review which we will assign when we have dev reviews completed.

File: i18n/en.json [hunk 1]
```diff
@@ -3085,13 +3085,13 @@
...
+  "rhs_header.closeSidebarTooltip": "Close",
+  "rhs_header.closeTooltip.icon": "Close Sidebar Icon",
   "rhs_header.details": "Message Details",
-  "rhs_header.expandSidebarTooltip": "Expand the sidebar",
-  "rhs_header.expandSidebarTooltip.icon": "Expand the sidebar icon",
-  "rhs_header.expandTooltip.icon": "Shrink the sidebar icon",
-  "rhs_header.shrinkSidebarTooltip": "Shrink the sidebar",
+  "rhs_header.expandSidebarTooltip": "Expand Sidebar",
+  "rhs_header.expandSidebarTooltip.icon": "Expand Sidebar Icon",
+  "rhs_header.expandTooltip.icon": "Collapse Sidebar Icon",
```
Comment (sudheerDev) [inline i18n/en.json:3093]: @esethna quick look for text changes here.

File: components/search_results_header/search_results_header.test.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            toggleRhsExpanded: jest.fn(),
+        },
+    };
+
+    test('should match snapshot', () => {
+        const wrapper = shallow(<SearchResultsHeader {...baseProps}/>);
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    const props = {
```
Comment (sudheerDev) [inline components/search_results_header/search_results_header.test.jsx:24]: Nit: Can you move this `const` declaration to the `test` scope below?

File: components/search_results_header/search_results_header.test.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        ...baseProps,
+        isExpanded: false
+    };
+
+    test('should match snapshot', () => {
```
Comment (sudheerDev) [inline components/search_results_header/search_results_header.test.jsx:29]: Can you change the title here to be `should match snapshot when collapsed` align with the functionality of the snapshot

**Comment by hmhealey** [comment]
@esethna Have you had a chance to double check the text changes?

**Comment by esethna** [comment]
@adityayavagal thanks for the PR. I cannot seem to get the tooltips to appear on the test server: https://mattermost-webapp-pr-5112.test.mattermost.cloud

image

**Comment by hmhealey** [comment]
That's strange. The OverlayTrigger doesn't have a `trigger` specified, but I would've assumed it defaulted to showing when hovered or focused. It's possible that it's being hidden behind the header, but it doesn't look like that based on some brief testing.

**Comment by adityaavysh** [comment]
well the tooltip should show when I hover on it right ? because I can see it when I hover on it

**Comment by adityaavysh** [comment]
If you give me your community server user name i can ping you there and validate if i did it right

**Comment by hmhealey** [comment]
Oh, I see the problem. The tooltip shows up when searching for a post, but not if you view a thread by clicking on the reply arrow on a post. Would you be able to add that as well? You'll want to look at the `RhsHeaderPost` component for that.

If you still want to chat on our community server though, my username is `@harrison` there

**Comment by adityaavysh** [comment]
alright i'll do that and check once

**Comment by adityaavysh** [comment]
Hey sorry, as I'm busy I'll need sometime but I'll do it.

**Comment by hmhealey** [comment]
No worries at all! Let us know if you have any more questions or could otherwise use some help

**Comment by ali-farooq0** [comment]
Hey @adityayavagal wondering if you'll able to get to this PR soon? Unfortunately I have a similar PR and wanted to see if you're able to fix up the changes so we can get this merged in?

Also, there are 2 other files that needs to be fixed as well:
```
components/rhs_card_header/rhs_card_header.jsx
components/rhs_header_post/rhs_header_post.jsx
```

**Comment by adityayavagal** [comment]
Ok give me some time I'll ping you today in community chat

**Comment by ali-farooq0** [comment]
Spoke to @adityayavagal offline -- Closing this this PR as my other [changes](https://github.com/mattermost/mattermost-webapp/pull/5626) are ready to go. Will work with @adityayavagal to see if he can grab another GH issue.

**Pull request closed without merging**